src/bp-core/admin/bp-core-admin-functions.php

```diff
@@ -3245,3 +3245,190 @@ function bp_core_get_moderation_admin_tabs( $active_tab = '' ) {
+			}
+
+			$app_preview_cover_url = $web_preview_cover_url;
+		} elseif ( ! $r['disable_cover_image_uploads'] && 'none' === $r['default_profile_cover_type'] ) {
+			$cover_class = ' has-cover';
+		}
+	} elseif ( 'group' === $object ) {
+
+		$r = bp_parse_args(
+			$_POST,
+			array(
+				'disable_avatar_uploads'      => bp_disable_group_avatar_uploads(),
+				'default_profile_avatar_type' => bb_get_default_group_avatar_type(),
+				'default_custom_avatar'       => bb_get_default_custom_upload_group_avatar( bb_get_buddyboss_group_avatar( $size ), $size ),
+				'disable_cover_image_uploads' => bp_disable_group_cover_image_uploads(),
+				'default_profile_cover_type'  => bb_get_default_group_cover_type(),
+				'default_custom_cover'        => bb_get_default_custom_upload_group_cover(),
+
+			),
+			$object . '_live_preview_urls_args'
+		);
+
+		// Group avatar
+		if ( ! $r['disable_avatar_uploads'] ) {
+
+			if ( 'buddyboss' === $r['default_profile_avatar_type'] ) {
+				$avatar_image_url = bb_get_buddyboss_group_avatar( $size );
+			} elseif ( 'legacy' === $r['default_profile_avatar_type'] ) {
+				$avatar_image_url = bb_get_legacy_group_avatar( $size );
+			} elseif ( 'custom' === $r['default_profile_avatar_type'] ) {
+				$avatar_image_url = ! empty( $r['default_custom_avatar'] ) ? $r['default_custom_avatar'] : bb_get_default_custom_upload_group_avatar( bb_get_buddyboss_group_avatar( $size ), $size );
+
+				if ( empty( $avatar_image_url ) ) {
+					$avatar_image_url = bb_get_buddyboss_group_avatar( $size );
+				}
+			}
+		}
+
+		// Group Cover
+		if ( ! $r['disable_cover_image_uploads'] && 'buddyboss' === $r['default_profile_cover_type'] ) {
+			$cover_class           = ' has-cover';
+			$web_preview_cover_url = buddypress()->plugin_url . 'bp-core/images/bb-cover-buddyboss-web.jpg';
+			$app_preview_cover_url = buddypress()->plugin_url . 'bp-core/images/bb-cover-buddyboss-app.jpg';
+		} elseif ( ! $r['disable_cover_image_uploads'] && 'custom' === $r['default_profile_cover_type'] ) {
+			$cover_class           = ' has-cover';
+			$web_preview_cover_url = ! empty( $r['default_custom_cover'] ) ? $r['default_custom_cover'] : bb_get_default_custom_upload_group_cover();
+
+			if ( empty( $web_preview_cover_url ) ) {
+				$web_preview_cover_url = bb_get_custom_buddyboss_group_cover();
+			}
+
+			$app_preview_cover_url = $web_preview_cover_url;
+		} elseif ( ! $r['disable_cover_image_uploads'] && 'none' === $r['default_profile_cover_type'] ) {
+			$cover_class = ' has-cover';
+		}
+	}
+
+	$response = array(
+		'avatar_class'   => ! empty( $avatar_image_url ) ? ' has-avatar' : '',
+		'cover_class'    => $cover_class,
+		'web_avatar_url' => $avatar_image_url,
+		'app_avatar_url' => $avatar_image_url,
+		'web_cover_url'  => $web_preview_cover_url,
+		'app_cover_url'  => $app_preview_cover_url,
+	);
+
+	wp_send_json_success( $response );
+}
+add_action( 'wp_ajax_bb_profile_group_get_live_preview_urls', 'bb_profile_group_get_live_preview_urls' );
```

Comment:
@milanhirapra I think we don't need this ajax to generate the preview box.
you can change the preview box using JS.